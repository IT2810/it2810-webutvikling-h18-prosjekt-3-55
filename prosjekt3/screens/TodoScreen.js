import React from 'react';
import { ScrollView,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Alert,
  AppRegistry,
  Text,
  View,
  FlatList,
  AsyncStorage,
  Button,
  TextInput,
  Keyboard,
  TouchableOpacity,
  Platform } from 'react-native';
import ScrollViewExample from '../components/TodoContainer';
import { Calendar, CalendarList, Agenda } from 'react-native-calendars';
import {store, retrieve} from "../components/Storage";


export default class TodoScreen extends React.Component {
  static navigationOptions = {
    title: 'Calendar',
  };

  static navigationOptions = {
    title: 'TODOs',
  };

      constructor(props) {
          super(props);
          this.state = {
              tasks: [],
              text: 'tester',
              todos: [
                  {'task': 'Ta medisinene dine', 'id': 1, 'checked': 0},
                  {'task': 'Puss tennene', 'id': 2, 'checked': 0},
                  {'task': 'Gi en god klem', 'id': 3, 'checked': 0},
                  {'task': 'Ha en nydelig dag', 'id': 4, 'checked': 0},],


          }


      }

  render() {


    return (
      <View >
      <Calendar onDayPress={(day) => {store(day.dateString, this.state.todos)} }
      />

          <View>
              <ScrollView>
                  {this.state.todos.map((item, index) => (
                      <View key = {item.id} style = {styles.item} >
                          <Text>{item.task}</Text>
                      </View>
                  ))}
              </ScrollView>
              <Text>  </Text>
          </View>
</View>


    );
  }
}
const styles = StyleSheet.create ({
    item: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'flex-start',
        padding: 15,
        margin: 2,
        borderColor: '#23B5D3',
        borderWidth: 1,
        backgroundColor: '#F7F7FF'
    }
})


