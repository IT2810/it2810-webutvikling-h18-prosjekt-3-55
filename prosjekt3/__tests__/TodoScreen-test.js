import 'react-native';
import React from 'react';
import TodoScreen from '../screens/TodoScreen';

import renderer from 'react-test-renderer';

test('snapshot homescreen', () => {
    const tree = renderer.create(<TodoScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});