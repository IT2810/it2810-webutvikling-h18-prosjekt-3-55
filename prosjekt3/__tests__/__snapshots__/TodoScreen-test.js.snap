// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot homescreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fbfbfb",
        "bottom": 0,
        "left": 0,
        "paddingVertical": 20,
        "position": "absolute",
        "right": 0,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": -3,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(96,100,109, 1)",
          "fontSize": 17,
          "textAlign": "center",
        }
      }
    >
      Txq:
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0.05)",
            "borderRadius": 3,
            "paddingHorizontal": 4,
          },
          Object {
            "marginTop": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(96,100,109, 0.8)",
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
      >
        navigation/MainTabNavigator.js
      </Text>
    </View>
  </View>
</View>
`;
